src/js/game/Entities/Ghast.js

```diff
@@ -87,26 +123,133 @@ module.exports = class Ghast extends BaseEntity {
-        case 0:
-          this.controller.audioPlayer.play("moan2");
-          break;
-        case 1:
-          this.controller.audioPlayer.play("moan3");
-          break;
-        case 2:
-          this.controller.audioPlayer.play("moan6");
-          break;
-        default:
-          this.controller.audioPlayer.play("moan7");
-          break;
+      this.audioDelay = 5;
+      let chance = Math.floor(Math.random() * 5);
+      if (chance === 0) {
+        let soundNum = Math.floor(Math.random() * 4);
+        this.playMoan(soundNum);
+      }
+    }
+  }
+
+  playMoan(number) {
+    switch (number) {
+      case 0:
+        this.controller.audioPlayer.play("moan2");
+        break;
+      case 1:
+        this.controller.audioPlayer.play("moan3");
+        break;
+      case 2:
+        this.controller.audioPlayer.play("moan6");
+        break;
+      default:
+        this.controller.audioPlayer.play("moan7");
+        break;
+    }
+  }
+
+// this.pattern = {type: <type>, specifics: <orientation or starting justification>};
+  fly() {
+    switch (this.details.pattern) {
+      case "box":
+        this.box(this.details.variant);
+        break;
+      case "sinusoidal":
+        this.sinusoidal(this.details.variant);
+        break;
+      default:
+        //do nothing
+    }
+  }
+
+  sinusoidal(variant) {
+    let mainAxis = null;
+    let offAxis = null;
+    if (variant === this.sinusoidalVariants.vertical) {
+      mainAxis = 'y';
+      offAxis = 'x';
+    } else if (variant === this.sinusoidalVariants.horizontal) {
+      mainAxis = 'x';
+      offAxis = 'y';
+    } else {
+      return;
+    }
+
+    if (!this.movingUp) { // flying along the main axis, positive direction
+      if (this.sprite.position[mainAxis] < this.flightAnchor[mainAxis] + (40 * this.primaryScalar)) {
+        this.sprite.position[mainAxis] += this.speed;
+      } else {
+        this.pauseTime += 1;
+        if (this.pauseTime > 0) {
+          this.pauseTime = 0;
+          this.movingUp = true;
         }
       }
+    } else { // Flying along the main axis, negative direction
+      if (this.sprite.position[mainAxis] > this.flightAnchor[mainAxis] - (40 * this.primaryScalar)) {
+        this.sprite.position[mainAxis] -= this.speed;
+      } else {
+        this.pauseTime += 1;
+        if (this.pauseTime > 0) {
+          this.pauseTime = 0;
+          this.movingUp = false;
+        }
+      }
+    }
+
+
+    this.sprite.position[offAxis] = this.flightAnchor[offAxis] + (Math.cos(this.trigCounter) * this.secondaryScalar);
+    this.trigCounter += this.speed/60;
+  }
+
+  box(corner) {
+    const vertices = this.getBoxVerts(corner);
+    let origin = this.boxVertTracker;
+    let destination = 0;
+    if (origin !== 3) {
+      destination = this.boxVertTracker + 1;
+    }
+
+    this.t += (this.speed / 60);
+    if (this.t > 1) {
+      this.t = 1;
     }
+    let pos = this.lerp(vertices[origin], vertices[destination], this.t);
```

Comment:
We should be leveraging Phaser tweens to move the sprite, rather than calculating the position ourselves.

Reply:
I did find it a bit absurd I had to write my own lerp function 😂 
Do phaser tweens have a lerp method that can be easily called? I can look into this for sure.